// react-native-tab-view/lib/module/TabView.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import * as React from 'react';
import { StyleSheet, View } from 'react-native';
import TabBar from './TabBar';
import SceneView from './SceneView';
import Pager from './Pager';
export default class TabView extends React.Component {
  constructor(...args) {
    super(...args);

    _defineProperty(this, "state", {
      layout: {
        width: 0,
        height: 0,
        ...this.props.initialLayout
      }
    });

    _defineProperty(this, "jumpToIndex", index => {
      if (index !== this.props.navigationState.index) {
        this.props.onIndexChange(index);
      }
    });

    _defineProperty(this, "handleLayout", e => {
      const {
        height,
        width
      } = e.nativeEvent.layout;

      if (this.state.layout.width === width && this.state.layout.height === height) {
        return;
      }

      this.setState({
        layout: {
          height,
          width
        }
      });
    });
  }

  render() {
    const {
      onSwipeStart,
      onSwipeEnd,
      navigationState,
      lazy,
      lazyPreloadDistance,
      keyboardDismissMode,
      swipeEnabled,
      tabBarPosition,
      renderTabBar,
      renderScene,
      renderLazyPlaceholder,
      sceneContainerStyle,
      style
    } = this.props;
    const {
      layout
    } = this.state;
    return /*#__PURE__*/React.createElement(View, {
      onLayout: this.handleLayout,
      style: [styles.pager, style]
    }, /*#__PURE__*/React.createElement(Pager, {
      layout: layout,
      navigationState: navigationState,
      keyboardDismissMode: keyboardDismissMode,
      swipeEnabled: swipeEnabled,
      onSwipeStart: onSwipeStart,
      onSwipeEnd: onSwipeEnd,
      onIndexChange: this.jumpToIndex
    }, ({
      position,
      render,
      addEnterListener,
      jumpTo
    }) => {
      // All of the props here must not change between re-renders
      // This is crucial to optimizing the routes with PureComponent
      const sceneRendererProps = {
        position,
        layout,
        jumpTo
      };
      return /*#__PURE__*/React.createElement(React.Fragment, null, tabBarPosition === 'top' && renderTabBar({ ...sceneRendererProps,
        navigationState
      }), render(navigationState.routes.map((route, i) => {
        return /*#__PURE__*/React.createElement(SceneView, _extends({}, sceneRendererProps, {
          addEnterListener: addEnterListener,
          key: route.key,
          index: i,
          lazy: typeof lazy === 'function' ? lazy({
            route
          }) : lazy,
          lazyPreloadDistance: lazyPreloadDistance,
          navigationState: navigationState,
          style: sceneContainerStyle
        }), ({
          loading
        }) => loading ? renderLazyPlaceholder({
          route
        }) : renderScene({ ...sceneRendererProps,
          route
        }));
      })), tabBarPosition === 'bottom' && renderTabBar({ ...sceneRendererProps,
        navigationState
      }));
    }));
  }

}

_defineProperty(TabView, "defaultProps", {
  tabBarPosition: 'top',
  renderTabBar: props => /*#__PURE__*/React.createElement(TabBar, props),
  renderLazyPlaceholder: () => null,
  keyboardDismissMode: 'auto',
  swipeEnabled: true,
  lazy: false,
  lazyPreloadDistance: 0,
  removeClippedSubviews: false
});

const styles = StyleSheet.create({
  pager: {
    flex: 1,
    overflow: 'hidden'
  }
});
