// react-native-tab-view/lib/commonjs/TabView.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _reactNative = require("react-native");

var _TabBar = _interopRequireDefault(require("./TabBar"));

var _SceneView = _interopRequireDefault(require("./SceneView"));

var _Pager = _interopRequireDefault(require("./Pager"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function () { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

class TabView extends React.Component {
  constructor(...args) {
    super(...args);

    _defineProperty(this, "state", {
      layout: {
        width: 0,
        height: 0,
        ...this.props.initialLayout
      }
    });

    _defineProperty(this, "jumpToIndex", index => {
      if (index !== this.props.navigationState.index) {
        this.props.onIndexChange(index);
      }
    });

    _defineProperty(this, "handleLayout", e => {
      const {
        height,
        width
      } = e.nativeEvent.layout;

      if (this.state.layout.width === width && this.state.layout.height === height) {
        return;
      }

      this.setState({
        layout: {
          height,
          width
        }
      });
    });
  }

  render() {
    const {
      onSwipeStart,
      onSwipeEnd,
      navigationState,
      lazy,
      lazyPreloadDistance,
      keyboardDismissMode,
      swipeEnabled,
      tabBarPosition,
      renderTabBar,
      renderScene,
      renderLazyPlaceholder,
      sceneContainerStyle,
      style
    } = this.props;
    const {
      layout
    } = this.state;
    return /*#__PURE__*/React.createElement(_reactNative.View, {
      onLayout: this.handleLayout,
      style: [styles.pager, style]
    }, /*#__PURE__*/React.createElement(_Pager.default, {
      layout: layout,
      navigationState: navigationState,
      keyboardDismissMode: keyboardDismissMode,
      swipeEnabled: swipeEnabled,
      onSwipeStart: onSwipeStart,
      onSwipeEnd: onSwipeEnd,
      onIndexChange: this.jumpToIndex
    }, ({
      position,
      render,
      addEnterListener,
      jumpTo
    }) => {
      // All of the props here must not change between re-renders
      // This is crucial to optimizing the routes with PureComponent
      const sceneRendererProps = {
        position,
        layout,
        jumpTo
      };
      return /*#__PURE__*/React.createElement(React.Fragment, null, tabBarPosition === 'top' && renderTabBar({ ...sceneRendererProps,
        navigationState
      }), render(navigationState.routes.map((route, i) => {
        return /*#__PURE__*/React.createElement(_SceneView.default, _extends({}, sceneRendererProps, {
          addEnterListener: addEnterListener,
          key: route.key,
          index: i,
          lazy: typeof lazy === 'function' ? lazy({
            route
          }) : lazy,
          lazyPreloadDistance: lazyPreloadDistance,
          navigationState: navigationState,
          style: sceneContainerStyle
        }), ({
          loading
        }) => loading ? renderLazyPlaceholder({
          route
        }) : renderScene({ ...sceneRendererProps,
          route
        }));
      })), tabBarPosition === 'bottom' && renderTabBar({ ...sceneRendererProps,
        navigationState
      }));
    }));
  }

}

exports.default = TabView;

_defineProperty(TabView, "defaultProps", {
  tabBarPosition: 'top',
  renderTabBar: props => /*#__PURE__*/React.createElement(_TabBar.default, props),
  renderLazyPlaceholder: () => null,
  keyboardDismissMode: 'auto',
  swipeEnabled: true,
  lazy: false,
  lazyPreloadDistance: 0,
  removeClippedSubviews: false
});

const styles = _reactNative.StyleSheet.create({
  pager: {
    flex: 1,
    overflow: 'hidden'
  }
});
